File: backend-py/app/schemas/article.py
from datetime import datetime

from pydantic import BaseModel

from app.models.article import ArticleStatus
from app.schemas.user import UserBrief
from app.schemas.category import CategoryResponse
from app.schemas.tag import TagResponse


class ArticleCreate(BaseModel):
    title: str
    content: str
    summary: str | None = None
    cover_image: str | None = None
    category_id: int
    tag_ids: list[int] | None = None
    status: ArticleStatus = ArticleStatus.DRAFT


class ArticleUpdate(BaseModel):
    title: str | None = None
    content: str | None = None
    summary: str | None = None
    cover_image: str | None = None
    category_id: int | None = None
    status: ArticleStatus | None = None
    tag_ids: list[int] | None = None


class ArticleBrief(BaseModel):
    id: int
    title: str
    slug: str
    summary: str | None = None
    cover_image: str | None = None
    status: ArticleStatus
    view_count: int
    created_at: datetime
    updated_at: datetime
    author: UserBrief
    category: CategoryResponse
    tags: list[TagResponse] = []

    model_config = {"from_attributes": True}


class ArticleResponse(ArticleBrief):
    content: str

File: admin/src/api/article.ts
import request from '@/utils/request'
import type { AdminArticle, ArticleStatus, PageResult, Category, Tag } from '@/types'

export const listArticles = (params: { page?: number; page_size?: number; status?: string; q?: string }) =>
  request.get<any, PageResult<AdminArticle>>('/admin/articles', { params })

export const getArticle = (id: number) =>
  request.get<any, ArticleDetail>(`/admin/articles/${id}`)

export const createArticle = (data: ArticleFormData) =>
  request.post<any, ArticleDetail>('/admin/articles', data)

export const updateArticle = (id: number, data: Partial<ArticleFormData>) =>
  request.put<any, ArticleDetail>(`/admin/articles/${id}`, data)

export const deleteArticle = (id: number) =>
  request.delete(`/admin/articles/${id}`)

export const reviewArticle = (id: number, status: ArticleStatus) =>
  request.put(`/admin/articles/${id}/review`, null, { params: { status } })

export const batchUpdateArticles = (ids: number[], status: ArticleStatus) =>
  request.post('/admin/articles/batch-update', ids, { params: { status } })

export const getCategories = () =>
  request.get<any, Category[]>('/admin/categories')

export const getTags = () =>
  request.get<any, Tag[]>('/admin/tags')

export interface ArticleFormData {
  title: string
  content: string
  summary?: string
  cover_image?: string
  category_id: number
  tag_ids?: number[]
  status?: ArticleStatus
}

export interface ArticleDetail extends AdminArticle {
  content: string
  summary: string | null
  cover_image: string | null
  author: { id: number; name: string; avatar: string | null } | null
  category: { id: number; name: string; slug: string } | null
  tags: { id: number; name: string; slug: string }[]
  created_at: string
  updated_at: string
}
